sql/driver/driver_test.go

```diff
@@ -335,35 +335,70 @@ CREATE TABLE t.kv (
 		t.Fatal(err)
 	}
 
-	rows, err := db.Query("SELECT * FROM t.kv")
-	if err != nil {
+	statementSuffixWithoutWhere := "FROM t.kv"
+	statementSuffixWithWhere := statementSuffixWithoutWhere + "  WHERE k IN ('a', 'c')"
+
+	// TODO(pmattis): We need much more testing of WHERE clauses. Need to think
+	// through the whole testing story in general.
+	if rows, err := db.Query("SELECT * " + statementSuffixWithWhere); err != nil {
 		t.Fatal(err)
+	} else {
+		results := readAll(t, rows)
+		expectedResults := [][]string{
+			{"k", "v"},
+			{"a", "b"},
+			{"c", "d"},
+		}
+		if !reflect.DeepEqual(expectedResults, results) {
+			t.Fatalf("expected %s, but got %s", expectedResults, results)
+		}
 	}
-	results := readAll(t, rows)
-	expectedResults := [][]string{
-		{"k", "v"},
-		{"a", "b"},
-		{"c", "d"},
-		{"e", "f"},
+
+	if rows, err := db.Query("SELECT * FROM t.kv"); err != nil {
+		t.Fatal(err)
+	} else {
+		results := readAll(t, rows)
+		expectedResults := [][]string{
+			{"k", "v"},
+			{"a", "b"},
+			{"c", "d"},
+			{"e", "f"},
+		}
+		if !reflect.DeepEqual(expectedResults, results) {
+			t.Fatalf("expected %s, but got %s", expectedResults, results)
+		}
 	}
-	if !reflect.DeepEqual(expectedResults, results) {
-		t.Fatalf("expected %s, but got %s", expectedResults, results)
+
+	if _, err := db.Exec("DELETE " + statementSuffixWithWhere); err != nil {
+		t.Fatal(err)
 	}
 
-	// TODO(pmattis): We need much more testing of WHERE clauses. Need to think
-	// through the whole testing story in general.
-	rows, err = db.Query("SELECT * FROM t.kv WHERE k IN ('a', 'c')")
-	if err != nil {
+	if rows, err := db.Query("SELECT * " + statementSuffixWithWhere); err != nil {
 		t.Fatal(err)
+	} else {
+		results := readAll(t, rows)
+		expectedResults := [][]string{
+			{"k", "v"},
+		}
+		if !reflect.DeepEqual(expectedResults, results) {
+			t.Fatalf("expected %s, but got %s", expectedResults, results)
+		}
 	}
-	results = readAll(t, rows)
-	expectedResults = [][]string{
-		{"k", "v"},
-		{"a", "b"},
-		{"c", "d"},
+
+	if _, err := db.Exec("DELETE " + statementSuffixWithoutWhere); err != nil {
+		t.Fatal(err)
 	}
-	if !reflect.DeepEqual(expectedResults, results) {
-		t.Fatalf("expected %s, but got %s", expectedResults, results)
+
+	if rows, err := db.Query("SELECT * " + statementSuffixWithoutWhere); err != nil {
+		t.Fatal(err)
+	} else {
+		results := readAll(t, rows)
+		expectedResults := [][]string{
+			{"k", "v"},
+		}
+		if !reflect.DeepEqual(expectedResults, results) {
+			t.Fatalf("expected %s, but got %s", expectedResults, results)
+		}
```

Comment:
You should add a new test for a multi-column primary key. Something with a table that looks like:

```
CREATE TABLE foo (
  a CHAR,
  b CHAR,
  c CHAR,
  PRIMARY KEY (a, b)
)
```